Sip Core is not getting user details in containerized HASS and backup_user is being sent as undefined to Asterisk

Hi,

After upgrade from beta 4.1.5 to 5.0.2 (ensuring all references where removed from all hass folders) now the integration is not working. In 4.1.5 it was working flawlessly.

What I have identified is that for some reason, in a containerized installation of Home Assistant, the integration is not able to get the logged in user, falling back to the backup_user which is not being registered into Asterisk as the request is being sent by undefined, instead of the extension configured.

<img width="1225" height="139" alt="Image" src="https://github.com/user-attachments/assets/8513d7c7-ff87-4ea3-9901-626aed0e7e17" />

<img width="3795" height="532" alt="Image" src="https://github.com/user-attachments/assets/b495512b-16e6-4da0-8ecb-1f52af4c56e3" />

And the configuration:

```
pbx_server: phone.xxx.xxxx.es
custom_wss_url: wss://phone.xxxxx.xxxx.es:8089/ws
outgoingRingtoneUrl: /sip_core_files/ringback-tone.mp3
incomingRingtoneUrl: /sip_core_files/ring-tone.mp3
backup_user:
  - ha_username: wallpanel
    extension: "103"
    password: "1234567"
users:
  - ha_username: sergio
    extension: "102"
    password: "1234567"
  - ha_username: wallpanel
    extension: "103"
    password: "1234567"
sip_video: false
auto_answer: false
popup_config:
  auto_open: true
  large: true
  hide_header_button: false
  buttons:
    - label: Open Door
      icon: mdi:door-open
      type: service_call
      data:
        domain: dahua
        service: vto_open_door
        door_id: "1"
        entity_id: camera.vto_main_2
  extensions:
    "103":
      name: Wallpanel
    "8001":
      name: VTO
      camera_entity: camera.vto_main_2
```

In 4.1.5 I had this configuration and it worked with no issue for the last 2-3 months:

```cat sip-config.json
{
    "custom_wss_url": "wss://phone.xxxx.xxxxx.es:8089/ws",
    "outgoingRingtoneUrl": "/local/out_ring-tone.mp3",
    "incomingRingtoneUrl": "/local/ringtone.mp3",
    "users": [
        {
            "ha_username": "sergio",
            "extension": "102",
            "password": "1234567"
        },
        {
            "ha_username": "wallpanel",
            "extension": "103",
            "password": "1234567"
        }
    ],
    "sip_video": false,
    "auto_answer": false,
    "popup_config": {
        "auto_open": true,
        "large": true,
        "hide_header_button": false,
        "buttons": [
            {
                "label": "Open Door 2",
                "icon": "mdi:door",
                "type": "service_call",
                "data": {
                    "domain": "dahua",
                    "service": "vto_open_door",
                    "door_id": 1,
                    "entity_id": "camera.vto_main_2"
                }
            }
        ],
        "extensions": {
            "103": {
                "name": "Wallpanel"
            },
            "8001": {
                "name": "VTO",
                "camera_entity": "camera.vto_main_2"
            }
        }
    }
}
```

Activating tracing into the developer console, the console logs the following:

```
sip-core.ts:344 SIP-Core Config fetched: {pbx_server: '10.69.2.4', custom_wss_url: 'wss://phone.xxxx.xxxxx.es:8089/ws', outgoingRingtoneUrl: '/sip_core_files/ringback-tone.mp3', incomingRingtoneUrl: '/sip_core_files/ring-tone.mp3', backup_user: Array(1), …}auto_answer: falsebackup_user: [{…}]custom_wss_url: "wss://phone.xxx.xxxx.es:8089/ws"incomingRingtoneUrl: "/sip_core_files/ring-tone.mp3"outgoingRingtoneUrl: "/sip_core_files/ringback-tone.mp3"pbx_server: "10.69.2.4"popup_config: {auto_open: true, large: true, hide_header_button: false, buttons: Array(1), extensions: {…}}sip_video: falseusers: (2) [{…}, {…}][[Prototype]]: Object
sip-core.ts:136 Using custom WSS URL: wss://phone.xxxx.xxxxx.es:8089/ws
sip-core.ts:599 Home Assistant Supervisor API not available. Assuming not running on Home Assistant OS.
sip-core.ts:327 Error fetching persons from Home Assistant: TypeError: Cannot read properties of undefined (reading 'id')
    at l.setupUser (sip-core.ts:324:109)
    at async l.init (sip-core.ts:303:20)
    at async sip-core.ts:630:21
setupUser @ sip-core.ts:327
await in setupUser
init @ sip-core.ts:304
await in init
(anonymous) @ sip-core.ts:631
o.a @ async module:49
886 @ browser.js:269
o @ bootstrap:19
(anonymous) @ Logger.js:41
o.a @ async module:49
156 @ Logger.js:41
o @ bootstrap:19
(anonymous) @ startup:4
(anonymous) @ startup:4
sip-core.ts:330 Selected user: undefined (undefined)
sip-core.ts:306 Connecting to wss://phone.xxxx.xxxxx.es:8089/ws...
sip-core.ts:236 Audio output set to null
sip-core.ts:249 Audio input set to null
sip-core.ts:436 Registration failed: {response: IncomingResponse, cause: 'Authentication Error'}
(anonymous) @ sip-core.ts:436
i.emit @ events.js:153
registrationFailed @ UA.js:558
_registrationFailure @ Registrator.js:419
onReceiveResponse @ Registrator.js:306
_receiveResponse @ RequestSender.js:166
onReceiveResponse @ RequestSender.js:56
receiveResponse @ Transactions.js:128
b @ UA.js:1114
_onData @ Transport.js:396
_onMessage @ WebSocketInterface.js:170
```

--- a/src/sip-core.ts
+++ b/src/sip-core.ts
@@ -319,12 +319,21 @@ export class SIPCore {
     }
 
     private async setupUser(): Promise<void> {
-        try {
-            const persons = await this.hass.callWS({ type: "person/list" });
-            const currentUsername = persons.storage.find((person: any) => person.user_id === this.hass.user.id).id;
-            this.user = this.config.users.find((user) => user.ha_username === currentUsername) || this.config.backup_user;
-        } catch (error) {
-            console.error("Error fetching persons from Home Assistant:", error);
+        const currentUserId = this.hass.user.id;
+        const currentUserName = this.hass.user.name;
+
+        // 1. Check exact match by User ID
+        let matchedUser = this.config.users.find((user) => user.ha_username === currentUserId);
+
+        // 2. Check match by User Name (if no ID match)
+        if (!matchedUser) {
+            matchedUser = this.config.users.find((user) => user.ha_username === currentUserName);
+        }
+
+        if (matchedUser) {
+            this.user = matchedUser;
+        } else {
+            console.warn(`No matching SIP user found for Home Assistant user: ${currentUserName} (${currentUserId}). Using backup user.`);
             this.user = this.config.backup_user;
         }
         console.debug(`Selected user: ${this.user.ha_username} (${this.user.extension})`);

